external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -270,17 +272,36 @@ public void setWaitingToEmit(ConsumerRecords<K,V> consumerRecords) {
 
     // ======== poll =========
     private ConsumerRecords<K, V> pollKafkaBroker() {
-        doSeekRetriableTopicPartitions();
+        Set<TopicPartition> retriableTopicPartitions = doSeekRetriableTopicPartitions();
+        Set<TopicPartition> pausedTopicPartitions = pauseNonRetriableTopicPartitionsIfAtUncommittedOffsetsLimit(retriableTopicPartitions);
         if (refreshSubscriptionTimer.isExpiredResetOnTrue()) {
             kafkaSpoutConfig.getSubscription().refreshAssignment();
         }
         final ConsumerRecords<K, V> consumerRecords = kafkaConsumer.poll(kafkaSpoutConfig.getPollTimeoutMs());
```

Comment:
Minor: I'd rather apply defensive programming here (to guarantee calling `resume` at any chance) like below: 
```
try {
    final ConsumerRecords<K, V> consumerRecords = kafkaConsumer.poll(kafkaSpoutConfig.getPollTimeoutMs());
    final int numPolledRecords = consumerRecords.count();
    LOG.debug("Polled [{}] records from Kafka. [{}] uncommitted offsets across all topic partitions", numPolledRecords, numUncommittedOffsets);
    return consumerRecords;
} finally {
    if (!pausedTopicPartitions.isEmpty()) {
        //Ensure that we resume all partitions in case some were paused when processing retries
        kafkaConsumer.resume(kafkaConsumer.assignment());
    }
}
```